src/pipecat/processors/user_marker_processor.py

```diff
@@ -0,0 +1,19 @@
+from pipecat.frames.frames import AudioRawFrame, Frame, UserAudioFrame
+from pipecat.processors.frame_processor import FrameDirection, FrameProcessor
+
+
+class UserMarkerProcessor(FrameProcessor):
+    """
+    This class extends FrameProcessor, used to mark the user's audio in the pipeline.
+    This FrameProcessor must be inserted after transport.input() so that the only
+    AudioRaw it receives are from the user.
+    """
```

Comment:
In theory this is not needed. All the audio coming from `transport.input()` is from the user. I believe what you have done in `AudioBufferProcessor` by checking `UserStartedSpeaking` and `UserStoppedSpeaking` should be enough.

Reply:
@aconchillo I can get this to work by only using `UserStartedSpeaking` and `UserStoppedSpeaking` but there is an longer than delay between when the assistant finishes and the human starts. This delay does not reflect what happened reality.  I'm pretty sure it's because audio from the user's mic is coming in and since there's no way to differentiate it, it gets added to the buffer while the assistant is talking. That is, the amount of extra delay added is the amount of time the assistant is speaking because all that silence from the user's mic is coming in and getting added to the buffer. There's a small mention of this in the discord, here: https://discord.com/channels/1239284677165056021/1278753324274814977/1279769874482204814

I'm happy to try an alternative implementation, but this is the best one I could come up with. The inflated delay is pretty bad, I don't think it would lead to usable feature.

Let me know your thoughts.

Reply:
So weird.... I'll take a look again.

Reply:
Just for others who find this.... The reason this happens is because of this `DailyParams`:

```
                vad_audio_passthrough=True,
                vad_analyzer=SileroVADAnalyzer(),
                transcription_enabled=True,
```

The STT will happen on the Daily server so really no audio would be required in the pipeline. However, since we want to save the audio we have `vad_audio_passthrough` which continuously pushes audio frames.